plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemColor.java

```diff
@@ -3,156 +3,69 @@
 import com.denizenscript.denizen.objects.ItemTag;
 import com.denizenscript.denizen.objects.properties.bukkit.BukkitColorExtensions;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ColorTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.tags.Attribute;
 import org.bukkit.Color;
-import org.bukkit.Material;
 import org.bukkit.inventory.meta.LeatherArmorMeta;
 import org.bukkit.inventory.meta.MapMeta;
 import org.bukkit.inventory.meta.PotionMeta;
 
-public class ItemColor implements Property {
+public class ItemColor extends ItemProperty<ColorTag> {
 
-    public static boolean describes(ObjectTag item) {
-        if (!(item instanceof ItemTag)) {
-            return false;
-        }
-        Material mat = ((ItemTag) item).getBukkitMaterial();
-        // Leather armor, potions, and filled map
-        return mat == Material.LEATHER_BOOTS
-                || mat == Material.LEATHER_CHESTPLATE
-                || mat == Material.LEATHER_HELMET
-                || mat == Material.LEATHER_LEGGINGS
-                || mat == Material.LEATHER_HORSE_ARMOR
-                || mat == Material.POTION
-                || mat == Material.SPLASH_POTION
-                || mat == Material.LINGERING_POTION
-                || mat == Material.FILLED_MAP;
-    }
+    // <--[property]
+    // @object ItemTag
+    // @name color
+    // @input ColorTag
+    // @description
+    // The color of a leather armor, potion, filled map, or tipped arrow item.
```

Comment:
Should mention that it can return null for maps, e.g. `For maps, will return null if the map doesn't have any color.`